src/main/kotlin/snyk/container/YAMLImageExtractor.kt

```diff
@@ -6,35 +6,41 @@ import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.psi.PsiFile
 import com.intellij.psi.PsiManager
 import java.util.stream.Collectors
-import kotlin.streams.toList
 
 object YAMLImageExtractor {
     private val logger = Logger.getInstance(javaClass.name)
 
-    private fun extractImages(file: PsiFile): List<String> {
-        return if (!isKubernetes(file)) emptyList() else extractImagesFromText(file)
+    private fun extractImages(file: PsiFile): Set<KubernetesWorkloadImage> {
+        return if (!isKubernetes(file)) emptySet() else extractImagesFromText(file)
     }
 
-    private fun extractImagesFromText(psiFile: PsiFile) =
-        getFileLines(psiFile).stream()
-            .filter { s -> s.trim().startsWith("image:") }
-            .map { s ->
-                s.trim()
-                    .split(":")
-                    .stream().filter { e -> e.trim() != "image" }
-                    .map { image -> image.trim() }
-                    .collect(Collectors.joining(":"))
-            }.toList()
+    private fun extractImagesFromText(psiFile: PsiFile): Set<KubernetesWorkloadImage> {
+        val extractedImages = mutableListOf<KubernetesWorkloadImage>()
+        getFileLines(psiFile).forEachIndexed { lineNumber, line ->
```

Comment:
current implementation of getFileLines() will fail on Win with `\n\r` line ends (lines will double as we'll have empty lines)